src/test/java/io/lettuce/SslTest.java

```diff
@@ -283,6 +450,22 @@ public void pubSubSslAndBreakConnection() throws Exception {
         connection2.getStatefulConnection().close();
     }
 
+
+    private static RedisURI.Builder sslURIBuilder(int portOffset) {
+        return RedisURI.Builder.redis(host(), sslPort(portOffset))
+                .withSsl(true);
+    }
+
+    private static RedisURI.Builder masterSlaveSSLURIBuilder(int portOffset) {
+        return sslURIBuilder(portOffset)
+                // TODO - why do the master/slave tests need this but the standalone ones don't???
```

Comment:
What happens if you omit this one?

Reply:
The `SslTest` take a looong time to run (like 400+ seconds vs the 5-10 that they're at after my current changes). There's probably a subtle difference with how the initial connections are setup and how long we wait on connection failures. I haven't figured it out yet, but I do think it should be fixed before we call this ready to merge.

Reply:
This is all that's left of the issues we've identified, so I'll try to figure this out today.

Reply:
Found it. Check out the [changes in StaticMasterSlaveTopologyProvider.java](https://github.com/lettuce-io/lettuce-core/pull/684/files#diff-7081f59597866c51272e8e6ba44922b5) for details.